### PR TITLE
Abstract property has `generics`

### PR DESCRIPTION
**Pull request by PerisicJere**

### CODE REVIEW

File: library/tests/fy_py_files/test_fy_py_files/abstract_property_with_generics/greeting_t.py
Comment (PerisicJere) [inline library/tests/fy_py_files/test_fy_py_files/abstract_property_with_generics/greeting_t.py:1]: generic

File: library/tests/fy_py_files/test_fy_py_files/abstract_property_with_generics/abc_fy.py.expected
Comment (PerisicJere) [inline library/tests/fy_py_files/test_fy_py_files/abstract_property_with_generics/abc_fy.py.expected:1]: expected test

File: library/tests/fy_py_files/test_fy_py_files/abstract_property_with_generics/abc_fy.py
Comment (PerisicJere) [inline library/tests/fy_py_files/test_fy_py_files/abstract_property_with_generics/abc_fy.py:1]: generated test

File: library/tests/fy_py_files/test_fy_py_abstract_property_with_generics.py
Comment (PerisicJere) [inline library/tests/fy_py_files/test_fy_py_abstract_property_with_generics.py:1]: test

File: library/src/fy_library/mixins/property/abstract_property_file_split/using_abstract_property_regex_fy.py [hunk 1]
```diff
@@ -27,7 +27,9 @@
 
 _ABSTRACT_PROPERTY_REGEX: Final = re.compile(
     rf"property\s+(?P<abstract_property_name>{FY_ENTITY_REGEX_STRING})"
-    rf"\s*:\s*(?P<return_type>{PYTHON_MULTI_ENTITY_REGEX_STRING})\s*$",
+    rf"(\[(?P<generic_type>{PYTHON_MULTI_ENTITY_REGEX_STRING})]:"
+    rf"|"
```
Comment (vinkobuble) [inline library/src/fy_library/mixins/property/abstract_property_file_split/using_abstract_property_regex_fy.py:31]:
```suggestion
    "|"
```

File: library/src/fy_library/mixins/property/abstract_property_file_split/using_abstract_property_regex_fy.py [hunk 2]
```diff
@@ -43,13 +45,18 @@ def _abstract_property_file_split(self) -> AbstractPropertyFileSplitModel:
...
+            len(abstract_property_file_split) == 6
         ), f"Abstract property file split length {len(abstract_property_file_split)} is invalid"
 
         abstract_property_file_split_model = AbstractPropertyFileSplitModel(
             user_imports=abstract_property_file_split[0],
             abstract_property_name=abstract_property_file_split[1],
-            property_type=abstract_property_file_split[2],
+            generics=abstract_property_file_split[3]
+            if abstract_property_file_split[3] is not None
+            else "",
```
Comment (vinkobuble) [inline library/src/fy_library/mixins/property/abstract_property_file_split/using_abstract_property_regex_fy.py:56]:
```suggestion
            generics=abstract_property_file_split[3] or abstract_property_file_split[4],
```

File: library/tests/fy_py_files/test_fy_py_files/abstract_property_with_generics/abc_fy.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+#  file, You can obtain one at https://mozilla.org/MPL/2.0/.
+"""fy
+from .greeting_t import GreetingT
+
+
+property abstract_property_with_generics[GreetingT]:
```
Comment (vinkobuble) [inline library/tests/fy_py_files/test_fy_py_files/abstract_property_with_generics/abc_fy.py:8]: Remove colon from the end of the statement.

File: library/src/fy_library/mixins/property/abstract_property_file_split/using_abstract_property_regex_fy.py [hunk 4]
```diff
@@ -43,13 +45,18 @@ def _abstract_property_file_split(self) -> AbstractPropertyFileSplitModel:
         abstract_property_file_split = _ABSTRACT_PROPERTY_REGEX.split(self._fy_code)
 
         assert (
-            len(abstract_property_file_split) == 4
+            len(abstract_property_file_split) == 6
         ), f"Abstract property file split length {len(abstract_property_file_split)} is invalid"
```
Comment (vinkobuble) [inline library/src/fy_library/mixins/property/abstract_property_file_split/using_abstract_property_regex_fy.py:49]:
```
assert (abstract_property_file_split[3] is None or abstract_property_file_split[4] is None) and (abstract_property_file_split[3] or abstract_property_file_split[4]) is not None, ">>>Explanation<<<"
```

**Review by vinkobuble — APPROVED** [review]

**Review by vinkobuble — CHANGES_REQUESTED** [review]
use `property_type_or_generic`

File: library/src/fy_library/mixins/property/abstract_property_file_split/using_abstract_property_regex_fy.py (hunk 2 above)
Comment (PerisicJere) [reply library/src/fy_library/mixins/property/abstract_property_file_split/using_abstract_property_regex_fy.py:56]: Added

File: library/src/fy_library/mixins/property/abstract_property_file_split/using_abstract_property_regex_fy.py (hunk 4 above)
Comment (PerisicJere) [reply library/src/fy_library/mixins/property/abstract_property_file_split/using_abstract_property_regex_fy.py:49]: Added

File: library/src/fy_library/mixins/property/abstract_property_file_split/using_abstract_property_regex_fy.py (hunk 1 above)
Comment (PerisicJere) [reply library/src/fy_library/mixins/property/abstract_property_file_split/using_abstract_property_regex_fy.py:31]: Changed

File: library/tests/fy_py_files/test_fy_py_files/abstract_property_with_generics/abc_fy.py (hunk 3 above)
Comment (PerisicJere) [reply library/tests/fy_py_files/test_fy_py_files/abstract_property_with_generics/abc_fy.py:8]: Removed

File: library/src/fy_library/mixins/property/abstract_property_file_split/using_abstract_property_regex_fy.py [hunk 5]
```diff
@@ -43,13 +45,25 @@ def _abstract_property_file_split(self) -> AbstractPropertyFileSplitModel:
...
+        assert (
+            (
+                abstract_property_file_split[2] is None
+                or abstract_property_file_split[3] is None
+            )
+            and (
+                abstract_property_file_split[2]
+                or abstract_property_file_split[3] is not None
+            )
+        ), "Abstract property requires exactly one generic or property type, not both or neither."
```
Comment (vinkobuble) [inline library/src/fy_library/mixins/property/abstract_property_file_split/using_abstract_property_regex_fy.py:60]:
```suggestion
        ), "Abstract property requires either generic or property type."
```

**Review by vinkobuble — APPROVED** [review]

**Pull request merged**